dsgrid/config/dataset_config.py

```diff
@@ -190,222 +198,325 @@ class GrowthRateType(DSGEnum):
-    )
-    data_schema_type: DataSchemaType = Field(
-        title="data_schema_type",
-        description="Discriminator for data schema",
-        options=DataSchemaType.format_for_docs(),
-    )
-    data_schema: Any = Field(
-        title="data_schema",
-        description="Schema (table layouts) used for writing out the dataset",
-    )
-    dataset_version: Optional[str] = Field(
-        title="dataset_version",
-        description="The version of the dataset.",
-    )
-    description: str = Field(
-        title="description",
-        description="A detailed description of the dataset.",
-    )
-    origin_creator: str = Field(
-        title="origin_creator",
-        description="Origin data creator's name (first and last)",
-    )
-    origin_organization: str = Field(
-        title="origin_organization",
-        description="Origin organization name, e.g., NREL",
-    )
-    origin_contributors: List[str] = Field(
-        title="origin_contributors",
-        description="List of origin data contributor's first and last names"
-        """ e.g., ["Harry Potter", "Ronald Weasley"]""",
-        default=[],
-    )
-    origin_project: str = Field(
-        title="origin_project",
-        description="Origin project name",
-        optional=True,
-    )
-    origin_date: str = Field(
-        title="origin_date",
-        description="Date the source data was generated",
-    )
-    origin_version: str = Field(
-        title="origin_version",
-        description="Version of the origin data",
-    )
-    source: str = Field(
-        title="source",
-        description="Source of the data (text description or link)",
-    )
-    data_classification: DataClassificationType = Field(
-        title="data_classification",
-        description="Data security classification (e.g., low, moderate, high)",
-        options=DataClassificationType.format_for_docs(),
-    )
-    tags: List[str] = Field(
-        title="source",
-        description="List of data tags",
-        default=[],
-    )
-    enable_unit_conversion: bool = Field(
-        default=True,
-        description="If the dataset uses its dimension mapping for the metric dimension to also "
-        "perform unit conversion, then this value should be false.",
-    )
+    ]
+    dataset_type: Annotated[
+        InputDatasetType,
+        Field(
+            title="dataset_type",
+            description="Input dataset type.",
+            json_schema_extra={
+                "options": InputDatasetType.format_for_docs(),
+            },
+        ),
+    ]
+    dataset_qualifier_metadata: Annotated[
+        Optional[Union[QuantityModel, GrowthRateModel]],
+        Field(
+            default=QuantityModel(dataset_qualifier_type="quantity"),
+            title="dataset_qualifier_metadata",
+            description="Additional metadata to include related to the dataset_qualifier",
+            discriminator="dataset_qualifier_type",
```

Comment:
This is an example of using a discriminator. I had to slightly reorganize the fields to make it work.